assets/src/components/site-scan-results/site-scan-sources-detail.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useContext, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { SiteScan } from '../site-scan-context-provider';
+import { getPluginSlugFromFile } from '../../common/helpers/get-plugin-slug-from-file';
+import ClipboardButton from '../clipboard-button';
+
+/**
+ * Site scan error source detail component.
+ *
+ * @param {Object} props      Component props.
+ * @param {string} props.slug Slug of plugin or theme.
+ */
+export function SiteScanSourcesDetail( {
+	slug,
+} ) {
+	const [ hasCopied, setHasCopied ] = useState( false );
+	const { scannableUrls } = useContext( SiteScan );
+
+	/**
+	 * Get list of source that belongs to current plugin/theme.
+	 *
+	 * @param {Array} sources List of source.
+	 *
+	 * @return {Array} List of source that of current plugin/theme.
+	 */
+	const getAllowedSources = ( sources ) => {
+		return sources.filter( ( source ) => {
+			return slug === getPluginSlugFromFile( source.name );
+		} );
+	};
+
+	/**
+	 * Get list of errors for current plugin/theme.
+	 *
+	 * @param {Array} validationErrors List of validation errors.
+	 *
+	 * @return {Array} List of validation errors for current plugin/theme.
+	 */
+	const getAllowedErrors = ( validationErrors ) => {
+		const errors = [];
+
+		for ( const validationError of validationErrors ) {
+			const sources = validationError.sources || [];
+			const allowedSources = getAllowedSources( sources );
+			if ( allowedSources && 0 < allowedSources.length ) {
+				const error = {
+					...validationError,
+					sources: allowedSources,
+				};
+
+				errors.push( error );
+			}
+		}
+
+		return errors;
+	};
+
+	const extensionScannableUrls = [];
+
+	for ( const scannableUrl of scannableUrls ) {
+		const validationErrors = scannableUrl.validation_errors || [];
+		const allowedErrors = getAllowedErrors( validationErrors );
+
+		if ( allowedErrors && 0 < allowedErrors.length ) {
+			const item = {
+				...scannableUrl,
+				validation_errors: allowedErrors,
+			};
+			extensionScannableUrls.push( item );
+		}
+	}
+
+	const jsonData = JSON.stringify( extensionScannableUrls, null, 4 );
```

Comment:
Since this is an expensive calculation, it'd be best wrap it with a `useMemo` hook (here's an example based on `map` and `filter` but `for .. of` loop could be used as originally proposed; also the `getAllowedErrors` function is not added to the hook's deps list since I assume it will be extracted out of the component body; instead `slug` would be added here):

 ```suggestion
	const jsonData = useMemo( () => {
		const extensionScannableUrls = scannableUrls.map( ( scannableUrl ) => {
			const validationErrors = scannableUrl.validation_errors || [];
			const allowedErrors = getAllowedErrors( validationErrors );
			
			if ( allowedErrors?.length > 0 ) {
				return {
					...scannableUrl,
					validation_errors: allowedErrors,
				};
			}
			
			return null;
		} ).filter( Boolean );

		return JSON.stringify( extensionScannableUrls, null, 4 );
	}, [ scannableUrls ] );
```